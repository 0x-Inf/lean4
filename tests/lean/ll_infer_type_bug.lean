def f : list nat → bool
| []      := ff
| (a::as) := a > 0 && f as

#check f._main._cstage2

mutual def f1, f2, f3, f4, f5
with f1 : nat → bool
| 0 := f3 0
| x := f2 x
with f2 : nat → bool
| 0 := f4 0
| x := f3 x
with f3 : nat → bool
| 0 := f5 0
| x := f4 x
with f4 : nat → bool
| 0     := f5 0
| (x+1) := f4 x
with f5 : nat → bool
| 0 := tt
| _ := ff

#check f1._main._cstage2
#check f2._main._cstage2
#check f3._main._cstage2
#check f4._main._cstage2
#check f5._main._cstage2
