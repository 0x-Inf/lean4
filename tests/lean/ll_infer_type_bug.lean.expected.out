f._cstage2 : _obj â†’ UInt8
